pkg/app/piped/executor/scriptrun/scriptrun.go

```diff
@@ -0,0 +1,68 @@
+package scriptrun
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/pipe-cd/pipecd/pkg/app/piped/executor"
+	"github.com/pipe-cd/pipecd/pkg/model"
+)
+
+type registerer interface {
+	Register(stage model.Stage, f executor.Factory) error
+	RegisterRollback(kind model.RollbackKind, f executor.Factory) error
+}
+
+type Executor struct {
+	executor.Input
+}
+
+func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
+	e.LogPersister.Infof("Start executing the script run stage")
+
+	opts := e.Input.StageConfig.ScriptRunStageOptions
+	if opts == nil {
+		e.LogPersister.Infof("option for script run stage not found")
+		return model.StageStatus_STAGE_FAILURE
+	}
+
+	if opts.Run == "" {
+		return model.StageStatus_STAGE_SUCCESS
+	}
+
+	envs := make([]string, 0, len(opts.Env))
+	for key, value := range opts.Env {
+		envs = append(envs, key+"="+value)
+	}
+
+	cmd := exec.Command("/bin/sh", "-l", "-c", opts.Run)
+	cmd.Env = append(os.Environ(), envs...)
+	cmd.Stdout = e.LogPersister
+	cmd.Stderr = e.LogPersister
+
+	e.LogPersister.Infof("executing script:")
+	e.LogPersister.Infof(opts.Run)
+
+	if err := cmd.Run(); err != nil {
```

Comment:
Should we printout the err as stage log?

Reply:
Applied ✅ 
[57c694e](https://github.com/pipe-cd/pipecd/pull/4720/commits/57c694e0272c75ae7c14bdaabfca2ab33869321d)